c2_utils/src/mfx_c2_param_storage.cpp

```diff
@@ -136,7 +142,7 @@ c2_status_t MfxC2ParamStorage::ConfigParam(const C2Param& param, bool component_
         {
             std::lock_guard<std::mutex> lock(m_valuesMutex);
             auto value_found = m_values.find(index);
-            if (value_found != m_values.end()) {
+            if (value_found == m_values.end()) {
```

Comment:
Why the logic changed?

Reply:
Some output parameters for decoder can be queried. With original logic, error message will be returned to framework.

Reply:
we have different param saved in different variables, m_values used to hold readonly param, it cannot be changed. So if framework want to change any param in m_values, plugin should report READ_ONLY error to framework, value_found == m_values.end() means the param not found in m_values, but it may saved in other place like m_paramOperations. So per my understanding the original logic should be right.

Reply:
I remember I also did this change before and then changed it back...